# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.cpp
# Commit: 80fed0cf581c
# Full Hash: 80fed0cf581ca575f6dc90714834650f156bf1e5
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-30 03:51:35
# Description:
#   Bug 1633204 - Live WindowContexts should keep BrowsingContextGroup alive, r=farre
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D72893
# ==============================================================================

diff -r 8c1e8723a695 -r 80fed0cf581c dom/ipc/WindowGlobalParent.cpp
--- a/dom/ipc/WindowGlobalParent.cpp	Wed Apr 29 18:37:37 2020 +0000
+++ b/dom/ipc/WindowGlobalParent.cpp	Wed Apr 29 18:49:10 2020 +0000
@@ -632,6 +632,14 @@
 }
 
 void WindowGlobalParent::ActorDestroy(ActorDestroyReason aWhy) {
+  // If there are any non-discarded nested contexts when this WindowContext is
+  // destroyed, tear them down.
+  nsTArray<RefPtr<dom::BrowsingContext>> toDiscard;
+  toDiscard.AppendElements(Children());
+  for (auto& context : toDiscard) {
+    context->Detach(/* aFromIPC */ true);
+  }
+
   // Note that our WindowContext has become discarded.
   WindowContext::Discard();
 
